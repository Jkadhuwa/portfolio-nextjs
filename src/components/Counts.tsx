import React from "react";

const Counts = () => {
  return (
    <div className="md:block mt-28 m-36">
      <div className=" border-gray-100 border " />
      <div className="grid grid-cols-4 gap-4 place-items-center p-16">
        <div className="flex flex-col text-center">
          <span className="text-gray-900 text-4xl font-medium">2+</span>
          <span className="text-gray-400 font-medium uppercase text-sm mt-2">
            Years of experience
          </span>
        </div>
        <div className="flex flex-col text-center">
          <span className="text-gray-900 text-4xl font-medium">12+</span>
          <span className="text-gray-400 font-medium uppercase text-sm mt-2">
            PROJECTS FINISHED
          </span>
        </div>
        <div className="flex flex-col text-center">
          <span className="text-gray-900 text-4xl font-medium">2+</span>
          <span className="text-gray-400 font-medium uppercase text-sm mt-2">
            Certifications
          </span>
        </div>
        <div className="flex flex-col text-center">
          <span className="text-gray-900 text-4xl font-medium">5+</span>
          <span className="text-gray-400 font-medium uppercase text-sm mt-2">
            Happy Clients
          </span>
        </div>
      </div>

      <div className="flex  border-gray-100 border" />
    </div>
  );
};

export default Counts;
