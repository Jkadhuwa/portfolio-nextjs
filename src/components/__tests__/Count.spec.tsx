import React from "react";
import "@testing-library/jest-dom";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import FontAwesomeIcon from "../../__mocks__/@fortawesome/fontAwesomeIcon";

import Counts from "@components/Counts";

afterEach(cleanup);
beforeAll(() => FontAwesomeIcon());
jest.mock("../../public/logo.png", () => {
  return {
    default: { src: "/public/logo.png", height: 100, width: 100 },
  };
});


const createWrapper = () => {
  return render(<Counts />);
};
describe("Test Count component", () => {
  describe("Snapshot Tests", () => {
    const wrapper = createWrapper();

    expect(wrapper).toMatchSnapshot();
  });

  describe("Count Unit Tests", () => {
    it("Should render count Component correctly", () => {
      createWrapper();
    });

    it("should have project count section", () => {
      createWrapper();
     expect(screen.getByText(/PROJECTS FINISHED/i)).toBeInTheDocument();
    });
      it("should have certifivcation count section", () => {
        createWrapper();
        expect(screen.getByText(/CERTIFICATIONS/i)).toBeInTheDocument();
      });
  });
});
