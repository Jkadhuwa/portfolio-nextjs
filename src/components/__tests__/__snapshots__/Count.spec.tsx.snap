// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="md:m-28"
      >
        <div
          class="border-gray-100 border "
        />
        <div
          class="grid p-8 space-y-8 md:space-y-0 md:grid-cols-4 md:gap-4 md:place-items-center md:p-16 "
        >
          <div
            class="flex flex-col text-center justify-center "
          >
            <span
              class="text-gray-900 text-4xl font-medium"
            >
              2+
            </span>
            <span
              class="text-gray-400 font-medium uppercase text-sm mt-2"
            >
              Years of experience
            </span>
          </div>
          <div
            class="flex flex-col text-center justify-center "
          >
            <span
              class="text-gray-900 text-4xl font-medium"
            >
              12+
            </span>
            <span
              class="text-gray-400 font-medium uppercase text-sm mt-2"
            >
              PROJECTS FINISHED
            </span>
          </div>
          <div
            class="flex flex-col text-center justify-center"
          >
            <span
              class=" text-gray-900 text-4xl font-medium "
            >
              2+
            </span>
            <span
              class="text-gray-400 font-medium uppercase text-sm mt-2 md:h-10 lg:h-5"
            >
              Certifications
            </span>
          </div>
          <div
            class="flex flex-col text-center justify-center "
          >
            <span
              class="text-gray-900 text-4xl font-medium"
            >
              5+
            </span>
            <span
              class="text-gray-400 font-medium uppercase text-sm mt-2 md:h-10 lg:h-5 "
            >
              Happy Clients
            </span>
          </div>
        </div>
        <div
          class="flex border-gray-100 border"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="md:m-28"
    >
      <div
        class="border-gray-100 border "
      />
      <div
        class="grid p-8 space-y-8 md:space-y-0 md:grid-cols-4 md:gap-4 md:place-items-center md:p-16 "
      >
        <div
          class="flex flex-col text-center justify-center "
        >
          <span
            class="text-gray-900 text-4xl font-medium"
          >
            2+
          </span>
          <span
            class="text-gray-400 font-medium uppercase text-sm mt-2"
          >
            Years of experience
          </span>
        </div>
        <div
          class="flex flex-col text-center justify-center "
        >
          <span
            class="text-gray-900 text-4xl font-medium"
          >
            12+
          </span>
          <span
            class="text-gray-400 font-medium uppercase text-sm mt-2"
          >
            PROJECTS FINISHED
          </span>
        </div>
        <div
          class="flex flex-col text-center justify-center"
        >
          <span
            class=" text-gray-900 text-4xl font-medium "
          >
            2+
          </span>
          <span
            class="text-gray-400 font-medium uppercase text-sm mt-2 md:h-10 lg:h-5"
          >
            Certifications
          </span>
        </div>
        <div
          class="flex flex-col text-center justify-center "
        >
          <span
            class="text-gray-900 text-4xl font-medium"
          >
            5+
          </span>
          <span
            class="text-gray-400 font-medium uppercase text-sm mt-2 md:h-10 lg:h-5 "
          >
            Happy Clients
          </span>
        </div>
      </div>
      <div
        class="flex border-gray-100 border"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
